src/osd/OSD.cc

```diff
@@ -9081,6 +9081,7 @@ void OSD::ShardedOpWQ::prune_pg_waiters(OSDMapRef osdmap, int whoami)
 	}
       }
       if (slot.to_process.empty() &&
+	  slot.num_running == 0 &&
```

Comment:
Don't we need to somehow invoke prune_pg_waiters() at a later time? I don't see a mechanism that actually cleans up.